Source/JavaScriptCore/runtime/StructureInlines.h

```diff
@@ -832,12 +832,10 @@ ALWAYS_INLINE bool Structure::canPerformFastPropertyEnumeration() const
     // FIXME: Indexed properties can be handled.
     // https://bugs.webkit.org/show_bug.cgi?id=185358
 
-    if (hasIndexedProperties(indexingType()))
+    if (!isTargetEmptyJSFinalObject && hasIndexedProperties(indexingType()))
```

Comment:
Why? We should always allow this.

Reply:
Updated. `canPerformFastPropertyEnumeration` is used in JSONObject.cpp. Maybe we can update it in the future patches. Extract `canPerformFastPropertyEnumerationCommon`.